prow/googlecloudbuild/client/client.go

```diff
@@ -0,0 +1,182 @@
+func NewClient(ctx context.Context, credentialFile string) (*Client, error) {
+	var opts []option.ClientOption
+	// Authenticating with key file if it's provided
+	if len(credentialFile) > 0 {
+		if _, err := os.Stat(credentialFile); err != nil {
+			return nil, err
+		}
+		opts = append(opts, option.WithCredentialsFile(credentialFile))
+	}
+
+	cbClient, err := cloudbuild.NewClient(ctx, opts...)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{interactor: cbClient}, nil
+}
+
+// GetBuild gets build by it's id
+func (c *Client) GetBuild(ctx context.Context, project, id string) (*cloudbuildpb.Build, error) {
+	return c.interactor.GetBuild(ctx, &cloudbuildpb.GetBuildRequest{
+		ProjectId: project,
+		Id:        id,
+	})
+}
+
+// ListBuildsByTag lists builds by tags
+func (c *Client) ListBuildsByTag(ctx context.Context, project string, tags []string) ([]*cloudbuildpb.Build, error) {
+	const pageSize = 50
+	var res []*cloudbuildpb.Build
+	var tagsFilters []string
+	for _, tag := range tags {
+		tagsFilters = append(tagsFilters, "tags="+tag)
+	}
+
+	iter := c.interactor.ListBuilds(ctx, &cloudbuildpb.ListBuildsRequest{
+		ProjectId: project,
+		PageSize:  pageSize,
+		Filter:    strings.Join(tagsFilters, " AND "),
+	})
+	// ListBuilds already fetches all, just need to do pagination
+	pager := iterator.NewPager(iter, pageSize, "")
+	for {
+		var buildsInPage []*cloudbuildpb.Build
+		nextPageToken, err := pager.NextPage(&buildsInPage)
+		if err != nil {
+			return nil, err
+		}
+		if buildsInPage != nil {
+			res = append(res, buildsInPage...)
+		}
+		if nextPageToken == "" {
+			break
+		}
+	}
+	return res, nil
+}
+
+// CreateBuild creates build and wait for the operation to complete.
+func (c *Client) CreateBuild(ctx context.Context, project string, bld *cloudbuildpb.Build) (*cloudbuildpb.Build, error) {
+	op, err := c.interactor.CreateBuild(ctx, &cloudbuildpb.CreateBuildRequest{
+		ProjectId: project,
+		Build:     bld,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// CreateBuild returns CreateBuildOperation, wait until the operation results in build.
+	// CreateBuildOperation contains `Wait` method, which however polls every minute, which is
+	// too slow. So use `wait.Poll` instead.
+	var triggered *cloudbuildpb.Build
+	if err := wait.Poll(100*time.Millisecond, 30*time.Second, func() (bool, error) {
+		triggered, err = op.Poll(ctx)
+		if err != nil {
+			return false, fmt.Errorf("failed to create build in project %s: %w", project, err)
+		}
+		// Poll surprisingly waits until the build completes before returning build,
+		// Metadata somehow does
+		meta, err := op.Metadata()
+		if err != nil {
+			return false, fmt.Errorf("failed to get metadata in project %s: %w", project, err)
+		}
+		triggered = meta.GetBuild()
+		return triggered != nil, nil
+	}); err != nil {
+		return nil, fmt.Errorf("failed waiting for build in project %s appear: %w", project, err)
+	}
+	return triggered, nil
+}
+
+// CancelBuild cancels build and wait for it
```

Comment:
Does it wait automatically or do we need to put in a `wait.Poll` call in here too like we did for CreateBuild?

Reply:
it does wait by itself